File: Leetcode/739. Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        Stack = []
        Answer = [0] * len(temperatures)
        for i in range(len(temperatures) - 1, -1, -1):
            while Stack and temperatures[i] >= temperatures[Stack[-1]]: 
                Stack.pop()
            if len(Stack) != 0: 
                Answer[i] = Stack[-1] - i
            Stack.append(i)
        return Answer

File: Leetcode/739. Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack <Integer> Stack = new Stack<>();
        int N = temperatures.length;
        Stack.push(N - 1);
        int[] Answer = new int[N];
        Answer[N - 1] = 0;
        for (int i = N - 2; i > -1; i--) {
            while (Stack.size() > 0 && temperatures[i] >= temperatures[Stack.peek()]) {
                Stack.pop();
            }
            if (Stack.size() != 0) 
                Answer[i] = Stack.peek() - i;
            else 
                Answer[i] = 0;
            Stack.push(i);
        }
        return Answer;
    }
}